File: project/7.EX_総合理解度チェック/java/java_ex7_ex_2/src/tents/Ex7_ex_2.java
package tents;

public class Ex7_ex_2 {

	public static void main(String[] args) {
		// TODO 自動生成されたメソッド・スタブ
		Tent[] ts = new Tent[4];
		
		ts[0] = new TentA();
		ts[1] = new TentB();
		ts[2] = new TentC();
		ts[3] = new TentD();
		
		for( Tent t : ts ) {
			if( t instanceof Waterproof ) System.out.println( t.getClass() );
		}

	}

}

File: project/7.EX_総合理解度チェック/py/py_ex7_ex_2/src/ex7_ex_2.py
'''
Created on 2023/05/15

@author: Keroichi.T
'''

from tents.tentA import TentA
from tents.tentB import TentB
from tents.tentC import TentC
from tents.tentD import TentD
from tents.tent import Waterproof

def main():
    ts = [ TentA(), TentB(), TentC(), TentD() ]
    
    for t in ts:
        if isinstance( t, Waterproof ): print( t.__class__ )

if __name__ == '__main__':
    main()